### PR TITLE
Use ASSIGN when using = inside braces

### PR DESCRIPTION
**Pull request by Harkishen-Singh**
Signed-off-by: Harkishen-Singh <harkishensingh@hotmail.com>

Fixes: #7904

### CODE REVIEW

**Comment by roidelapluie** [comment]
Why should it be assign when we make actually a comparison? {a="n",a="m"} seems equals, not assign

**Comment by Harkishen-Singh** [comment]
As far as I know, we use `=` only in vector selectors. I don't think there is any case where `==` is used inside vector selectors. So, even though we call it equality, but in sense of tokens, it should be `=` which in our terms is ASSIGN.

**Comment by codesome** [comment]
If we keep aside EQL and ASSIGN for a while, I would expect a lex item to be called the same thing irrespective of where it appears (and lexer being unaware of the language grammar). `=` being called EQL in one place and ASSIGN in another does not sound right.

**Comment by Harkishen-Singh** [comment]
I think, `=` should be EQL and `==` as EQLC which stands for equal and compare. I don't think there is use of ASSIGN as a term.

**Comment by codesome** [comment]
@Harkishen-Singh sounds good. Can you make that change?

**Comment by Harkishen-Singh** [comment]
@codesome done!

**Comment by brian-brazil** [comment]
:+1:

File: promql/parser/generated_parser.y [hunk 1]
```diff
@@ -229,7 +229,7 @@ aggregate_modifier:
 // Operator precedence only works if each of those is listed separately.
 binary_expr     : expr ADD     bin_modifier expr { $$ = yylex.(*parser).newBinaryExpression($1, $2, $3, $4) }
                 | expr DIV     bin_modifier expr { $$ = yylex.(*parser).newBinaryExpression($1, $2, $3, $4) }
-                | expr EQL     bin_modifier expr { $$ = yylex.(*parser).newBinaryExpression($1, $2, $3, $4) }
+                | expr EQLC     bin_modifier expr { $$ = yylex.(*parser).newBinaryExpression($1, $2, $3, $4) }
```
Comment (codesome) [inline promql/parser/generated_parser.y:232]: Nit
```suggestion
                | expr EQLC    bin_modifier expr { $$ = yylex.(*parser).newBinaryExpression($1, $2, $3, $4) }
```

**Review by codesome — APPROVED** [review]

**Comment by Harkishen-Singh** [comment]
Done!

**Comment by codesome** [comment]
Thanks!

**Pull request merged**